nupic/research/frameworks/continuous_learning/multihead/subdataset.py

```diff
@@ -0,0 +1,191 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Dataset functionality for MNIST and Google Speech Commands in a continual learning
+framework.
+"""
+
+import os
+
+import numpy as np
+from torch.utils.data import Dataset
+from torchvision import transforms
+from torchvision.datasets import MNIST
+
+
+def get_datasets(dataset_name, scenario, download=False):
+    """
+    Return the train and test datasets (in parallel lists) for each task in the
+    continuous learning setup
+    :param dataset_name: name of the dataset to retrieve
+    :type dataset_name: str
+    :param scenario: continuous learning setup, one of {'task', 'domain', 'class'}
```

Comment:
This function doesn't distinguish between task or class, it only cares whether it is domain or not, which could be represented as a boolean flag to simplify the interface.

Reply:
I made the change you suggested, and modified all other modules that call this function.